plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -38,14 +39,19 @@ import qualified PlutusIR.Transform.ThunkRecursions as ThunkRec
 import           PlutusIR.TypeCheck.Internal
 
 import qualified PlutusCore                         as PLC
+import qualified PlutusCore.Constant.Meaning        as M
 
 import           Control.Monad
 import           Control.Monad.Reader
 import           PlutusPrelude
 
+-- | Actual simplifier
+simplify :: M.ToBuiltinMeaning uni fun => Term TyName Name uni fun b -> Term TyName Name uni fun b
+simplify = Beta.beta . Inline.inline . DeadCode.removeDeadBindings
```

Comment:
Doesn't it make more sense to do beta *before* the inliner? Since it actually *creates* let-bindings, which the inliner can deal with. There's no benefit to doing it afterwards - unless we simplify again later, but even then the last beta run will provide no benefit.

In fact, doing it before dead code makes sense, since dead code also uses let-bindings.